File: 프로그래머스/python/LV2/주식가격.py
from collections import deque


def solution(prices):
    result = []
    queue = deque(prices)

    while queue:
        time = 0
        price = queue.popleft()

        for q in queue:
            time += 1
            if price > q:
                break
        result.append(time)

    return result

File: 프로그래머스/javascript/LV2/주식가격.js
function solution(prices) {
    let arr = [];
    for(let i=0;i<prices.length;i++){
        let count = 0;
        for(let j=i+1;j<prices.length;j++){
            count++;
            if(prices[i]>prices[j]) break;
        }
        arr.push(count);
    }
    
    return arr;
}